src/signer/canister/tests/it/utils/pic_canister.rs

```diff
@@ -0,0 +1,279 @@
+            .map_err(|e| {
+                format!(
+                    "Query call error. RejectionCode: {:?}, Error: {}",
+                    e.code, e.description
+                )
+            })
+            .and_then(|reply| match reply {
+                WasmResult::Reply(reply) => {
+                    decode_one(&reply).map_err(|_| "Decoding failed".to_string())
+                }
+                WasmResult::Reject(error) => Err(error),
+            })
+    }
+    fn workspace_dir() -> PathBuf {
+        let output = std::process::Command::new(env!("CARGO"))
+            .arg("locate-project")
+            .arg("--workspace")
+            .arg("--message-format=plain")
+            .output()
+            .unwrap()
+            .stdout;
+        let cargo_path = Path::new(std::str::from_utf8(&output).unwrap().trim());
+        cargo_path.parent().unwrap().to_path_buf()
+    }
+    /// The path to a typical Cargo Wasm build.
+    #[allow(dead_code)]
+    fn cargo_wasm_path(name: &str) -> String {
+        let workspace_dir = Self::workspace_dir();
+        workspace_dir
+            .join("target/wasm32-unknown-unknown/release")
+            .join(name)
+            .with_extension("wasm")
+            .to_str()
+            .unwrap()
+            .to_string()
+    }
+    /// The path to the wasm after `dfx deploy`.  Expects the Wasm to be gzipped.
+    ///
+    /// If not already gzipped, please add this to the canister declaration in `dfx.json`: `"gzip": true`
+    #[allow(dead_code)]
+    fn dfx_wasm_path(name: &str) -> String {
+        Self::workspace_dir()
+            .join(format!(".dfx/local/canisters/{name}/{name}.wasm.gz"))
+            .to_str()
+            .unwrap()
+            .to_string()
+    }
+}
+
+/// A typical canister running on PocketIC.
+pub struct PicCanister {
+    pub pic: Arc<PocketIc>,
+    pub canister_id: Principal,
+}
+
+impl PicCanisterTrait for PicCanister {
+    /// The shared PocketIc instance.
+    fn pic(&self) -> Arc<PocketIc> {
+        self.pic.clone()
+    }
+    /// The ID of this canister.
+    fn canister_id(&self) -> Principal {
+        self.canister_id.clone()
+    }
+}
+
+impl PicCanister {
+    /// Creates a new canister.
+    pub fn new(pic: Arc<PocketIc>, wasm_path: &str) -> Self {
+        PicCanisterBuilder::default()
+            .with_wasm(wasm_path)
+            .deploy_to(pic)
+    }
+}
+
+/// Canister installer, using the builder pattern, for use in test environmens using `PocketIC`.
+///
+/// # Example
+/// For a default test environment:
+/// ```
+/// let pic_canister = BackendBuilder::default().deploy();
+/// ```
+/// To add a backend canister to an existing `PocketIC`:
+/// ```
+/// let pic = PocketIc::new();
+/// let canister_id = BackendBuilder::default().deploy_to(&pic);
+/// ```
+/// To redeploy an existing canister:
+/// ```
+/// // First deployment:
+/// let (pic, canister_id) = BackendBuilder::default().deploy();
```

Comment:
That it returns a `pic` is what confuses me a bit. Why not always expect the `&pic` as parameter?

The pic is the replica, right?

Reply:
And I also don't see the `deploy` function anywhere in this file. Is it something you'll add later or do yuo mean `deploy_to`?